Source/ACE/Entity/Player.cs

```diff
@@ -1823,7 +1918,56 @@ public void HandleActionSmiteSelected()
                 }
             }).EnqueueChain();
         }
-        
+
+        public void EquipUnequipItem(uint itemId, uint slot)
```

Comment:
This is public, and called from outside the player's action context (from a game message).  It should be HandleActionEquipUnequip, and use an ActionChain.